mirage/views/resource.ts

```diff
@@ -1,10 +1,25 @@
 import { HandlerContext, Request, Response, Schema } from 'ember-cli-mirage';
 
+type Undef = 'undefined';
+
+export function createResource(this: HandlerContext, schema: Schema) {
+    const attrs = {
+        ...this.normalizedRequestAttrs('resource'),
+        finalized: false,
+        resourceType: 'undefined' as Undef,
+    };
+    return schema.resources.create(attrs);
+}
+
 export function updateResource(this: HandlerContext, schema: Schema, request: Request) {
     const resource = schema.resources.find(request.params.id);
-    const { data: { attributes } } = JSON.parse(request.requestBody);
-    if ('pid' in attributes && (!attributes.pid || !attributes.pid.startsWith('https://doi.org/'))) {
-        return new Response(400);
+    const attributes = {
+        ...this.normalizedRequestAttrs('resource'),
+    };
+    if ('pid' in attributes) {
+        if (!attributes.pid || !attributes.pid.startsWith('https://doi.org/')) {
```

Comment:
This will have to change in the near future due to https://www.notion.so/cos/Handle-DOIs-as-non-uri-form-1b4eebc7017a4936966c43848b8a32e6 but it's probably fine to keep it like this for now.